pkg/proxy/userspace/roundrobin_test.go

```diff
@@ -97,6 +74,40 @@ func expectEndpointWithSessionAffinityReset(t *testing.T, loadBalancer *LoadBala
 	}
 }
 
+func TestBuildPortsToEndpointsMap(t *testing.T) {
+	service := proxy.ServicePortName{NamespacedName: types.NamespacedName{Namespace: "testnamespace", Name: "foo"}, Port: "p"}
+	endpoints := &v1.Endpoints{
+		ObjectMeta: metav1.ObjectMeta{Name: service.Name, Namespace: service.Namespace},
+		Subsets: []v1.EndpointSubset{{
+			Addresses: []v1.EndpointAddress{{IP: "endpoint1"}},
+			Ports:     []v1.EndpointPort{{Name: "p", Port: 40}},
+		},
+			{
+				Addresses: []v1.EndpointAddress{{IP: "endpoint2"}},
+				Ports:     []v1.EndpointPort{{Name: "p", Port: 80}},
+			}},
+	}
+	portsToEndpoints := util.BuildPortsToEndpointsMap(endpoints)
+	if len(portsToEndpoints["p"]) == 0 {
+		t.Errorf("portsToEndpoints map doesn't contain expected endpoint name")
+	}
+	endpts := portsToEndpoints["p"]
+	endptMap := map[string]bool{}
+	endptMap["endpoint1:40"] = false
+	endptMap["endpoint2:80"] = false
+	if len(endpts) != len(endptMap) {
```

Comment:
So for the testing, I'd create a map that matches what you expect to get back from `BuildPortsToEndpointsMap`, and then use `reflect.DeepEqual` to compare that the two maps are equal.

A few more test cases would also be good